e2e/test/helpers/TestDeviceCallbackHandler.cs

```diff
@@ -14,35 +13,42 @@ namespace Microsoft.Azure.Devices.E2ETests.Helpers
     internal sealed class TestDeviceCallbackHandler : IDisposable
     {
         private readonly IotHubDeviceClient _deviceClient;
-        private readonly TestDevice _testDevice;
+        private readonly string _testDeviceId;
 
         private readonly SemaphoreSlim _methodCallbackSemaphore = new(0, 1);
         private ExceptionDispatchInfo _methodExceptionDispatch;
+        private DirectMethodServiceRequest _expectedDirectMethodRequest;
 
         private readonly SemaphoreSlim _twinCallbackSemaphore = new(0, 1);
         private ExceptionDispatchInfo _twinExceptionDispatch;
-        private string _expectedTwinPropertyValue;
 
         private readonly SemaphoreSlim _receivedMessageCallbackSemaphore = new(0, 1);
         private ExceptionDispatchInfo _receiveMessageExceptionDispatch;
-        private OutgoingMessage _expectedMessageSentByService;
+        private OutgoingMessage _expectedOutgoingMessage
+            ;
 
-        internal TestDeviceCallbackHandler(IotHubDeviceClient deviceClient, TestDevice testDevice)
+        internal TestDeviceCallbackHandler(IotHubDeviceClient deviceClient, string deviceId)
         {
+            _testDeviceId = deviceId;
+
+            if (deviceClient == null || deviceClient.ConnectionStatusInfo.Status != ConnectionStatus.Connected)
```

Comment:
Our callbacks in this class were each opening the client. Instead, making it required for the client to be open when the handler is initialized.
We no longer have implicit open, so this is a reasonable expectation.